plugins/dql-backend/src/service/queryExecutor.ts

```diff
@@ -63,6 +69,56 @@ export class QueryExecutor {
     return results.flatMap(result => result);
   }
 
+  async executeCustomNotebookQueries(
+    notebookVariables: ComponentNotebookVariables,
+    variables: ComponentQueryVariables,
+  ): Promise<NotebookQueryData[] | undefined> {
+    componentQueryVariablesSchema.parse(variables);
+    if (this.apis.length > 1 && !notebookVariables.notebookUrl) {
+      throw new Error(
+        `The annotation notebook-id is only supported in the context of a single Dynatrace environment`,
+      );
+    }
+
+    const filteredApis = notebookVariables.notebookUrl
+      ? this.apis.filter(
+          api => api.config.url === 'https://' + notebookVariables.notebookUrl,
```

Comment:
This will not work if the config URL as trailing slashes right? Could you also take care that trailing slashes are allowed? Just to prevent https://github.com/Dynatrace/backstage-plugin/issues/99 from re-appearing

Reply:
I may be overlooking something but the current implementation is not considering an optional trailing slash of the API config URL right? At the moment the API may have a trailing slash and the notebookUrl will never have a trailing slash

Reply:
Thank you, yes you are right!